Makefile

```diff
@@ -68,10 +68,16 @@ checklicenses: .venv requirements.txt ## check dependencies meet licence rules
 	poetry run liccheck -s liccheck.ini
 
 test: .venv ## run tests quickly with the default Python
-	poetry run pytest tests/test_libgsf_load_valid.py --verbose --capture=no
-	poetry run pytest tests/test_libgsf_load_invalid.py --verbose --capture=no
-	poetry run pytest tests/test_libgsf_load_default.py --verbose --capture=no
-	poetry run pytest --ignore-glob=tests/test_libgsf_load_*.py --verbose --capture=no
+	poetry run pytest --ignore-glob=tests/gsfpy3_08/* --ignore-glob=tests/gsfpy3_09/* --ignore-glob=tests/gsfpy/test_gsffile_with_*.py --verbose --capture=no
+	poetry run pytest --ignore-glob=tests/gsfpy3_08/* --ignore-glob=tests/gsfpy3_09/* --ignore-glob=tests/gsfpy/test_gsffile.py --verbose --capture=no
+	poetry run pytest tests/gsfpy3_08/test_libgsf_load_valid.py --verbose --capture=no
+	poetry run pytest tests/gsfpy3_08/test_libgsf_load_invalid.py --verbose --capture=no
+	poetry run pytest tests/gsfpy3_08/test_libgsf_load_default.py --verbose --capture=no
+	poetry run pytest --ignore-glob=tests/gsfpy3_08/test_libgsf_load_*.py --ignore-glob=tests/gsfpy3_09/* --verbose --capture=no --cov=gsfpy3_08 --cov-fail-under=95 --cov-config=tox.ini
+	poetry run pytest tests/gsfpy3_09/test_libgsf_load_valid.py --verbose --capture=no
+	poetry run pytest tests/gsfpy3_09/test_libgsf_load_invalid.py --verbose --capture=no
+	poetry run pytest tests/gsfpy3_09/test_libgsf_load_default.py --verbose --capture=no
+	poetry run pytest --ignore-glob=tests/gsfpy3_09/test_libgsf_load_*.py --ignore-glob=tests/gsfpy3_08/* --verbose --capture=no --cov=gsfpy3_09 --cov-fail-under=95 --cov-config=tox.ini
```

Comment:
Is there a reason these are run individually rather than just pointing pytest at the tests folder?

Reply:
There is a reason! Once a module is loaded in python it is almost impossible to unload it. Therefore for tests which are checking the gsfpy behaviour of loading the correct version-specific namespace based on the DEFAULT_GSF_VERSION env variable, or loading custom versions of the libgsf shared object libraries, it is necessary to execute these in their own dedicated pytest runs.